AU2/plugins/custom_plugins/CompetencyPlugin.py

```diff
@@ -139,26 +170,71 @@ def on_hook_respond(self, hook: str, htmlResponse, data) -> List[HTMLComponent]:
-                        title="Extend competency?",
-                        default={},
-                        global_default=GENERIC_STATE_DATABASE.arb_int_state.get(self.plugin_state["DEFAULT"], DEFAULT_EXTENSION)
-                    )
-                ]
-            )
-        ]
+        if GENERIC_STATE_DATABASE.arb_state.get(self.plugin_state["Auto Competency"], "Manual") != "Full Auto":
+            return [
+                Dependency(
+                    dependentOn="CorePlugin_assassin_pseudonym",
+                    htmlComponents=[
+                        AssassinDependentIntegerEntry(
+                            pseudonym_list_identifier="CorePlugin_assassin_pseudonym",
+                            identifier=self.html_ids["Competency"],
+                            title="Extend competency?",
+                            default={},
+                            global_default=GENERIC_STATE_DATABASE.arb_int_state.get(self.plugin_state["DEFAULT"], DEFAULT_EXTENSION)
+                        )
+                    ]
+                )
+            ]
+        else:
+            return []
 
     def on_event_create(self, e: Event, htmlResponse) -> List[HTMLComponent]:
-        e.pluginState.setdefault(self.identifier, {})[self.plugin_state["COMPETENCY"]] = htmlResponse[self.html_ids["Competency"]]
-        return [Label("[COMPETENCY] Success!")]
+        message = []
+        competency_extension = GENERIC_STATE_DATABASE.arb_int_state.get(self.plugin_state["DEFAULT"], DEFAULT_EXTENSION)
+        if GENERIC_STATE_DATABASE.arb_state.get(self.plugin_state["Auto Competency"], "Manual") != "Full Auto":
+            e.pluginState.setdefault(self.identifier, {})[self.plugin_state["COMPETENCY"]] = htmlResponse[self.html_ids["Competency"]]
+        else:  # Create an empty competency dict when nothing was entered (due to full auto mode)
+            e.pluginState.setdefault(self.identifier, {})[self.plugin_state["COMPETENCY"]] = {}
+
+        if GENERIC_STATE_DATABASE.arb_state.get(self.plugin_state["Auto Competency"], "Manual") != "Manual":
+            # auto kill competency:
+            for (killer, victim) in e.kills:
+                if not ASSASSINS_DATABASE.get(victim).is_police:
+                    if killer not in e.pluginState[self.identifier][self.plugin_state["COMPETENCY"]]:
+                        # if to avoid overwriting manual entries
+                        message.append(Label(f"[COMPETENCY] Granted {competency_extension} days competency to {killer} for a kill"))
+                        e.pluginState[self.identifier][self.plugin_state["COMPETENCY"]][killer] = competency_extension
+
+            # auto attempt competency:
+            if GENERIC_STATE_DATABASE.plugin_map.get("AttemptPlugin", False):
+                # Is this reliable? It works, but seems to depend on AttemptPlugin.on_event_create() running before this
+                # If not, how would I do this?

```

Comment:
If it relies on plugin call ordering, that is determined by a hash-map ordering at initialization time so I wouldn't rely on this.

Two options:
1) The `CompetencyManager` can (and probably should) track this.
2) Merge the `AttemptPlugin` into the `CompetencyPlugin`

I'm partial to (1) - if you delete an earlier event after registering attempts on a later one, the second approach (or the current approach) don't update later events. Not that it matters much, but it's better complexity wise as you aren't looping over all events on adding every event, but tracking `attempts_since_last_kill` dynamically.

This resolves the comment you have later about issues with modifying.

Reply:
Another problem with (1) I realised as I'm implementing it:
Player gets kill (or 2 attempts), giving 7 days competency.
6 days later, umpires reduce comp time to 5 days.
Player is now inco, if competencymanager tries to dynamically calculate competency.
Should I revert to (2), which comes with the inability to deal with event edits/deletes?
Is there a way of doing everything without storing default competency changes in an event? Or should I just compromise.

Reply:
Good point. 

Store the current competency period alongside each attempt. When dynamically calculating it, you'll have full information.